homeassistant/config_entries.py

```diff
@@ -201,12 +201,12 @@ class ExampleConfigFlow(config_entries.ConfigFlow):
 class ConfigEntry:
     """Hold a configuration entry."""
 
-    __slots__ = ('entry_id', 'version', 'domain', 'title', 'data', 'source',
-                 'connection_class', 'state', '_setup_lock',
+    __slots__ = ('entry_id', 'version', 'domain', 'title', 'data', 'options',
+                 'source', 'connection_class', 'state', '_setup_lock',
                  '_async_cancel_retry_setup')
 
     def __init__(self, version: str, domain: str, title: str, data: dict,
-                 source: str, connection_class: str,
+                 source: str, connection_class: str, options: dict = {},
```

Comment:
Do we want to use a mutable object as default argument value?

https://docs.python-guide.org/writing/gotchas/

Probably not.

Reply:
If we want to make sure that this is a dict we should do that in the class body not in the signature.

Reply:
Eg:
```py
self.options = options or {}
```